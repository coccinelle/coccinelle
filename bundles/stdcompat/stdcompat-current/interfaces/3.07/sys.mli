val argv : string array
val executable_name : string
external file_exists : string -> bool = "sys_file_exists"
external remove : string -> unit = "sys_remove"
external rename : string -> string -> unit = "sys_rename"
external getenv : string -> string = "sys_getenv"
external command : string -> int = "sys_system_command"
external time : unit -> float = "sys_time"
external chdir : string -> unit = "sys_chdir"
external getcwd : unit -> string = "sys_getcwd"
external readdir : string -> string array = "sys_read_directory"
val interactive : bool ref
val os_type : string
val word_size : int
val max_string_length : int
val max_array_length : int
type signal_behavior =
  | Signal_default 
  | Signal_ignore 
  | Signal_handle of (int -> unit) 
external signal :
  int -> signal_behavior -> signal_behavior = "install_signal_handler"
val set_signal : int -> signal_behavior -> unit
val sigabrt : int
val sigalrm : int
val sigfpe : int
val sighup : int
val sigill : int
val sigint : int
val sigkill : int
val sigpipe : int
val sigquit : int
val sigsegv : int
val sigterm : int
val sigusr1 : int
val sigusr2 : int
val sigchld : int
val sigcont : int
val sigstop : int
val sigtstp : int
val sigttin : int
val sigttou : int
val sigvtalrm : int
val sigprof : int
exception Break 
val catch_break : bool -> unit
val ocaml_version : string
