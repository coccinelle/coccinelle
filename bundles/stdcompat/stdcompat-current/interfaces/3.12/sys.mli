val argv : string array
val executable_name : string
external file_exists : string -> bool = "caml_sys_file_exists"
external is_directory : string -> bool = "caml_sys_is_directory"
external remove : string -> unit = "caml_sys_remove"
external rename : string -> string -> unit = "caml_sys_rename"
external getenv : string -> string = "caml_sys_getenv"
external command : string -> int = "caml_sys_system_command"
external time : unit -> float = "caml_sys_time"
external chdir : string -> unit = "caml_sys_chdir"
external getcwd : unit -> string = "caml_sys_getcwd"
external readdir : string -> string array = "caml_sys_read_directory"
val interactive : bool ref
val os_type : string
val word_size : int
val max_string_length : int
val max_array_length : int
type signal_behavior =
  | Signal_default 
  | Signal_ignore 
  | Signal_handle of (int -> unit) 
external signal :
  int -> signal_behavior -> signal_behavior = "caml_install_signal_handler"
val set_signal : int -> signal_behavior -> unit
val sigabrt : int
val sigalrm : int
val sigfpe : int
val sighup : int
val sigill : int
val sigint : int
val sigkill : int
val sigpipe : int
val sigquit : int
val sigsegv : int
val sigterm : int
val sigusr1 : int
val sigusr2 : int
val sigchld : int
val sigcont : int
val sigstop : int
val sigtstp : int
val sigttin : int
val sigttou : int
val sigvtalrm : int
val sigprof : int
exception Break 
val catch_break : bool -> unit
val ocaml_version : string
