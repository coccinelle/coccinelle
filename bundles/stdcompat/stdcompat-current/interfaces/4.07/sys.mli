val argv : string array
val executable_name : string
external file_exists : string -> bool = "caml_sys_file_exists"
external is_directory : string -> bool = "caml_sys_is_directory"
external remove : string -> unit = "caml_sys_remove"
external rename : string -> string -> unit = "caml_sys_rename"
external getenv : string -> string = "caml_sys_getenv"
val getenv_opt : string -> string option
external command : string -> int = "caml_sys_system_command"
external time :
  unit -> ((float)[@unboxed ]) = "caml_sys_time" "caml_sys_time_unboxed"
[@@noalloc ]
external chdir : string -> unit = "caml_sys_chdir"
external getcwd : unit -> string = "caml_sys_getcwd"
external readdir : string -> string array = "caml_sys_read_directory"
val interactive : bool ref
val os_type : string
type backend_type =
  | Native 
  | Bytecode 
  | Other of string 
val backend_type : backend_type
val unix : bool
val win32 : bool
val cygwin : bool
val word_size : int
val int_size : int
val big_endian : bool
val max_string_length : int
val max_array_length : int
external runtime_variant : unit -> string = "caml_runtime_variant"
external runtime_parameters : unit -> string = "caml_runtime_parameters"
type signal_behavior =
  | Signal_default 
  | Signal_ignore 
  | Signal_handle of (int -> unit) 
external signal :
  int -> signal_behavior -> signal_behavior = "caml_install_signal_handler"
val set_signal : int -> signal_behavior -> unit
val sigabrt : int
val sigalrm : int
val sigfpe : int
val sighup : int
val sigill : int
val sigint : int
val sigkill : int
val sigpipe : int
val sigquit : int
val sigsegv : int
val sigterm : int
val sigusr1 : int
val sigusr2 : int
val sigchld : int
val sigcont : int
val sigstop : int
val sigtstp : int
val sigttin : int
val sigttou : int
val sigvtalrm : int
val sigprof : int
val sigbus : int
val sigpoll : int
val sigsys : int
val sigtrap : int
val sigurg : int
val sigxcpu : int
val sigxfsz : int
exception Break 
val catch_break : bool -> unit
val ocaml_version : string
val enable_runtime_warnings : bool -> unit
val runtime_warnings_enabled : unit -> bool
external opaque_identity : 'a -> 'a = "%opaque"
