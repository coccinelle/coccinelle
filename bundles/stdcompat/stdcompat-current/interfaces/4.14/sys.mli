external argv : string array = "%sys_argv"
val executable_name : string
external file_exists : string -> bool = "caml_sys_file_exists"
external is_directory : string -> bool = "caml_sys_is_directory"
external remove : string -> unit = "caml_sys_remove"
external rename : string -> string -> unit = "caml_sys_rename"
external getenv : string -> string = "caml_sys_getenv"
val getenv_opt : string -> string option
external command : string -> int = "caml_sys_system_command"
external time :
  unit -> ((float)[@unboxed ]) = "caml_sys_time" "caml_sys_time_unboxed"
[@@noalloc ]
external chdir : string -> unit = "caml_sys_chdir"
external mkdir : string -> int -> unit = "caml_sys_mkdir"
external rmdir : string -> unit = "caml_sys_rmdir"
external getcwd : unit -> string = "caml_sys_getcwd"
external readdir : string -> string array = "caml_sys_read_directory"
val interactive : bool ref
val os_type : string
type backend_type =
  | Native 
  | Bytecode 
  | Other of string 
val backend_type : backend_type
val unix : bool
val win32 : bool
val cygwin : bool
val word_size : int
val int_size : int
val big_endian : bool
val max_string_length : int
val max_array_length : int
val max_floatarray_length : int
external runtime_variant : unit -> string = "caml_runtime_variant"
external runtime_parameters : unit -> string = "caml_runtime_parameters"
type signal_behavior =
  | Signal_default 
  | Signal_ignore 
  | Signal_handle of (int -> unit) 
external signal :
  int -> signal_behavior -> signal_behavior = "caml_install_signal_handler"
val set_signal : int -> signal_behavior -> unit
val sigabrt : int
val sigalrm : int
val sigfpe : int
val sighup : int
val sigill : int
val sigint : int
val sigkill : int
val sigpipe : int
val sigquit : int
val sigsegv : int
val sigterm : int
val sigusr1 : int
val sigusr2 : int
val sigchld : int
val sigcont : int
val sigstop : int
val sigtstp : int
val sigttin : int
val sigttou : int
val sigvtalrm : int
val sigprof : int
val sigbus : int
val sigpoll : int
val sigsys : int
val sigtrap : int
val sigurg : int
val sigxcpu : int
val sigxfsz : int
exception Break 
val catch_break : bool -> unit
val ocaml_version : string
val development_version : bool
type extra_prefix =
  | Plus 
  | Tilde 
type extra_info = (extra_prefix * string)
type ocaml_release_info =
  {
  major: int ;
  minor: int ;
  patchlevel: int ;
  extra: extra_info option }
val ocaml_release : ocaml_release_info
val enable_runtime_warnings : bool -> unit
val runtime_warnings_enabled : unit -> bool
external opaque_identity : 'a -> 'a = "%opaque"
module Immediate64 :
sig
  module type Non_immediate  = sig type t end
  module type Immediate  = sig type t[@@immediate ] end
  module Make :
  functor (Immediate : Immediate) ->
    functor (Non_immediate : Non_immediate) ->
      sig
        type t[@@immediate64 ]
        type 'a repr =
          | Immediate: Immediate.t repr 
          | Non_immediate: Non_immediate.t repr 
        val repr : t repr
      end
end
