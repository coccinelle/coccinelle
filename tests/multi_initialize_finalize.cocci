/* i3 comes before i2 since ocaml is initialized in one step. */

@initialize:ocaml@
@@

let x = 0
let () = print_endline "i1"

@initialize:python@
@@

import sys
x = 0
print("i2")
sys.stdout.flush()

@initialize:ocaml@
@@

let y = 1
let () = print_endline "i3"

@initialize:python@
@@

y = 1
print("i4")
sys.stdout.flush()

@@
@@
- return 0;
+ return 1;

@finalize:ocaml@
@@
assert (x = 0);
print_endline "f1"

@finalize:python@
@@
assert x == 0
print("f2")
sys.stdout.flush()

@finalize:ocaml@
@@
assert (y = 1);
print_endline "f3"

@finalize:python@
@@
assert y == 1
print("f4")
sys.stdout.flush()