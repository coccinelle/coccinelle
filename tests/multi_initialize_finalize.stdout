i1
i3
i2
i4
f1
f2
f3
f4
